src/github.com/getlantern/flashlight/pac.go

```diff
@@ -41,16 +44,33 @@ func setUpPacTool() error {
 	return nil
 }
 
+func setProxyAddr(addr string) {
```

Comment:
In fact, an interesting project would be an extension-based version of detour.  In addition to the potential performance benefit, it would also allow for doing detection on HTTPS sites.